dm/syncer/safe_mode_test.go

```diff
@@ -80,21 +94,48 @@ func TestEnableSafeModeInitializationPhase(t *testing.T) {
 	// test enable by config
 	s.cliArgs = nil
 	s.cfg.SafeMode = true
+	s.cfg.SafeModeDuration = "0" // test safeMode's priority higher than SafeModeDuration's
 	mockCheckpoint := &mockCheckpointForSafeMode{}
+	mockCheckpoint.globalPoint = binlog.Location{}
+	mockCheckpoint.tablePoint = make(map[string]map[string]binlog.Location)
 	s.checkpoint = mockCheckpoint
 	s.enableSafeModeInitializationPhase(s.tctx)
 	require.True(t, s.safeMode.Enable())
```

Comment:
but in fact this case (safe-mode = true & safe-mode-duration = 0s) should not happen?

Reply:
In short, `safe-mode` has a high priority to `safe-mode-duration`. So `safe-mode-duration = 0s` doesn't affect `safe-mode = true`.

Reply:
> but in fact this case (safe-mode = true & safe-mode-duration = 0s) should not happen?

This case may not happen.

The test case is just testing the priority.